src/auxiliar.h

```diff
@@ -31,7 +31,9 @@
 
 #include "luasocket.h"
 
+#ifndef _WIN32
```

Comment:
Shouldn't the `#ifndef` be GCC-specific?

Here is what I use in similar situations:
```C
#if (defined( __GNUC__ ) && __GNUC__ >= 4) || __has_attribute( __visibility__ )
```

You would need a fallback for all compilers that don't handle `__has_attribute`:
```C
#ifndef __has_attribute
#  define __has_attribute( x ) 0
#endif
```

Reply:
Although you probably have to handle `_WIN32 ` (-> MinGW) separately anyway ...